core/src/main/resources/lib/layout/header/actions.jelly

```diff
@@ -20,17 +20,47 @@
 
     <!--  we only want the hamburger menu if there is something to display -->
     <!--  render all non primary non current actions -->
+    <j:set var="hamburgerHasBadgeWithSeverityError" value="false" />
+    <j:set var="hamburgerHasBadgeWithSeverityWarning" value="false" />
+    <j:set var="hamburgerHasBadgeWithSeverityInfo" value="false" />
     <j:set var="hamburgerEntries">
       <j:forEach var="action" items="${allActions}">
         <j:set var="isCurrent" value="${h.hyperlinkMatchesCurrentPage(action.urlName)}" />
         <j:set var="isPrimary" value="${h.showInPrimaryHeader(action)}"/>
         <j:if test="${!isCurrent and !isPrimary}">
           <h:secondaryAction action="${action}"/>
+          <j:set var="currentBadge" value="${action.badge}"/>
+          <j:if test="${currentBadge != null}">
+            <j:choose>
+              <j:when test="${currentBadge.severity == 'danger'}">
+                <j:set var="hamburgerHasBadgeWithSeverityError" value="true" />
+              </j:when>
+              <j:when test="${currentBadge.severity == 'warning'}">
+                <j:set var="hamburgerHasBadgeWithSeverityWarning" value="true" />
+              </j:when>
+              <j:otherwise>
+                <j:set var="hamburgerHasBadgeWithSeverityInfo" value="true" />
+              </j:otherwise>
+            </j:choose>
+          </j:if>
         </j:if>
       </j:forEach>
     </j:set>
     <j:if test="${hamburgerEntries.length() gt 0}">
-      <l:overflowButton icon="symbol-menu-hamburger" id="header-more-actions">
+      <j:set var="hamburgerBadge">
+        <j:choose>
+          <j:when test="${hamburgerHasBadgeWithSeverityError}">
+            <span class="jenkins-badge jenkins-!-danger-color" />
```

Comment:
this actually reveals a UX issue with the current design that is out of scope of this PR as it does exactly what is already done in the rest of the header.
Namely the visual indication of severity is only differentiated by colour.  A colour blind user can not easily tell at a glance what the severity of the action is.